hack/fake-apiserver/README.md

```diff
@@ -0,0 +1,8 @@
+# CAPM3 Fake API server
```

Comment:
So this says CAPM3 Fake API server, L4 points to CAPIM (which is coherent and refer to a provider), but L5 says "kubernetes servers" and also the directory name is "fake-apiserver" which both associate to "k8s apiserver". 

So based on that readme, I have no idea if this is faking the k8s apiserver or CAPM3 or what. Please reword/rename the readme and the directory so its clear what it is faking.